src/Campaigns/Controllers/CampaignRequestController.php

```diff
@@ -31,7 +32,11 @@ public function getCampaign(WP_REST_Request $request)
             return new WP_Error('campaign_not_found', __('Campaign not found', 'give'), ['status' => 404]);
         }
 
-        return new WP_REST_Response($campaign->toArray());
+        return new WP_REST_Response(
+            array_merge($campaign->toArray(), [
+                'goalProgress' => (new CampaignDonationQuery($campaign))->sumIntendedAmount(),
+            ])
+        );
```

Comment:
@kjohnson If we're adding `goalProgress` to the resource payload, then we need to add it as a readonly property to the object schema. This will tell WP that it can be read but not included in write (POST, PUT, PATCH) requests.

It will look something like:
```json
"properties": {
	"goalProgress": {
		"type": "number",
		"readonly": true
	}
}
```

Reply:
Resolved, see [f0b6ff3](https://github.com/impress-org/givewp/pull/7628/commits/f0b6ff38fd3f4f55447f91653645f3563855e976).